interfaces/builtin/all.go

```diff
@@ -21,72 +21,82 @@ package builtin
 
 import (
 	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/release"
 )
 
-var allInterfaces = []interfaces.Interface{
-	&BluezInterface{},
-	&BoolFileInterface{},
-	&BrowserSupportInterface{},
-	&ContentInterface{},
-	&DockerInterface{},
-	&DockerSupportInterface{},
-	&FwupdInterface{},
-	&GpioInterface{},
-	&HidrawInterface{},
-	&I2cInterface{},
-	&LocationControlInterface{},
-	&LocationObserveInterface{},
-	&LxdInterface{},
-	&LxdSupportInterface{},
-	&MirInterface{},
-	&ModemManagerInterface{},
-	&MprisInterface{},
-	&NetworkManagerInterface{},
-	&OfonoInterface{},
-	&PppInterface{},
-	&PulseAudioInterface{},
-	&SerialPortInterface{},
-	&UDisks2Interface{},
-	NewAlsaInterface(),
-	NewAvahiObserveInterface(),
-	NewBluetoothControlInterface(),
-	NewCameraInterface(),
-	NewCupsControlInterface(),
-	NewDcdbasControlInterface(),
-	NewFirewallControlInterface(),
-	NewFuseSupportInterface(),
-	NewGsettingsInterface(),
-	NewHardwareObserveInterface(),
-	NewHomeInterface(),
-	NewKernelModuleControlInterface(),
-	NewLibvirtInterface(),
-	NewLocaleControlInterface(),
-	NewLogObserveInterface(),
-	NewMountObserveInterface(),
-	NewNetworkBindInterface(),
-	NewNetworkControlInterface(),
-	NewNetworkInterface(),
-	NewNetworkObserveInterface(),
-	NewNetworkSetupObserveInterface(),
-	NewOpenglInterface(),
-	NewOpticalDriveInterface(),
-	NewProcessControlInterface(),
-	NewRawUsbInterface(),
-	NewRemovableMediaInterface(),
-	NewScreenInhibitControlInterface(),
-	NewShutdownInterface(),
-	NewSnapdControlInterface(),
-	NewSystemObserveInterface(),
-	NewSystemTraceInterface(),
-	NewTimeControlInterface(),
-	NewTimeserverControlInterface(),
-	NewTimezoneControlInterface(),
-	NewTpmInterface(),
-	NewUPowerObserveInterface(),
-	NewUnity7Interface(),
-	NewX11Interface(),
+func initInterfaces() []interfaces.Interface {
```

Comment:
Can we avoid changing this?  Ideally `allInterfaces` would remain being _all interfaces._ so that snapd has metadata about all the interfaces it supports, even if it's unsupported in the local system.